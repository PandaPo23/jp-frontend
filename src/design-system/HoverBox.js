import PropTypes from 'prop-types';

import dynamicVariant from '../utils/DynamicVariant';
import system from '../utils/System';
import Box from './Box';

const hoverVariant = dynamicVariant({
  key: 'hovers',
  prop: 'hover',
});

const HoverBox = system(
  'HoverBox',
  {
    extend: Box,
    bg: 'surface',
    color: 'on.surface',
    hover: 'bg.darken',
  },
  hoverVariant
);

HoverBox.propTypes = {
  hover: PropTypes.string,
};

export default HoverBox;
