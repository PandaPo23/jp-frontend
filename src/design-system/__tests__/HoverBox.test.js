import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import HoverBox from '../HoverBox';

describe('HoverBox component', () => {
  it('should render as HoverBox', () => {
    const wrapper = mount(<HoverBox />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
