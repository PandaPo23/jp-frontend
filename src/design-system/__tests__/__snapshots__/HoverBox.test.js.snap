// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoverBox component should render as HoverBox 1`] = `
.emotion-0 {
  color: on.surface;
  background-color: surface;
}

<HoverBox
  bg="surface"
  color="on.surface"
  hover="bg.darken"
>
  <div
    className="emotion-0"
  />
</HoverBox>
`;
